File: CRM_admin/lib/models/news.dart
class News{
  final String id;
  final String title;
  final String payload;
  final DateTime datePublished;

  News(this.title, this.payload, this.datePublished, this.id);

  factory News.fromMap(Map<String, dynamic> map) {
    return News(
      map["title"],
      map["payload"],
      DateTime.parse(map["datePublished"]),
      map["id"]
    );
  }
}

File: CRM_server/Models/News.cs
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson;

namespace Server.Models
{
    public class News
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }

        [BsonElement("title")]
        [BsonRequired]
        public string Title { get; set; } = null!;

        [BsonElement("payload")]
        [BsonRequired]
        public string Payload { get; set; } = null!;

        [BsonElement("datePublished")]
        [BsonRequired]
        public DateTime DatePublished { get; set; }
    }
}
